lib/vrl/compiler/benches/path.rs

```diff
@@ -0,0 +1,45 @@
+use criterion::{criterion_group, criterion_main, BenchmarkId, Criterion, Throughput};
+use std::fmt;
+use std::str::FromStr;
+use vrl_compiler::path::Path;
+
+struct Parameters {
+    basis: &'static str,
+}
+
+static PARAMETERS: [Parameters; 3] = [
+    Parameters {
+        basis: ".foo.(bar | baz)[1].(qux | quux)",
+    },
+    Parameters {
+        basis: ".foo.(bar | baz)[1].(qux | quux).(a|b)",
+    },
+    Parameters {
+        basis: ".foo.(bar | baz)[1].(qux | quux).(a|b).(c|d)",
+    },
+];
+
+impl fmt::Display for Parameters {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", self.basis)
+    }
+}
+
+fn benchmark_to_alternative_strings(c: &mut Criterion) {
+    let mut group = c.benchmark_group("path::to_alternative_strings");
+    for param in &PARAMETERS {
+        group.throughput(Throughput::Bytes(param.basis.len() as u64));
+
+        group.bench_with_input(BenchmarkId::from_parameter(param), &param, |b, &param| {
+            let path = Path::from_str(param.basis).unwrap();
+            b.iter(|| {
+                let _ = path.to_alternative_strings();
```

Comment:
```suggestion
                path.to_alternative_strings()
```

I'm unsure if the compiler actually does this, but I could see it optimizing that call away. Returning the value causes criterion to wrap it in `black_box`.

Reply:
Incorporated.